plugins/inputs/system/hugepages.go

```diff
@@ -0,0 +1,184 @@
+package system
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var (
+	newlineByte     = []byte("\n")
+	colonByte       = []byte(":")
+	kbPrecisionByte = []byte("kB")
+)
+
+// default file paths
+const (
+	// the path where statistics are kept per NUMA nodes
+	NUMA_NODE_PATH = "/sys/devices/system/node"
+	// the path to the meminfo file which is produced by kernel
+	MEMINFO_PATH = "/proc/meminfo"
+
+	// hugepages stat field names on meminfo file
+	HUGE_PAGES_TOTAL = "HugePages_Total"
+	HUGE_PAGES_FREE  = "HugePages_Free"
+)
+
+var hugepagesSampleConfig = `
+  ## Path to a NUMA nodes
+  # numa_node_path = "/sys/devices/system/node"
+  ## Path to a meminfo file
+  # meminfo_path = "/proc/meminfo"
+`
+
+// Mem is the
+type Hugepages struct {
+	NUMANodePath string `toml:"numa_node_path"`
+	MeminfoPath  string `toml:"meminfo_path"`
+}
+
+func (mem *Hugepages) Description() string {
+	return "Collects hugepages metrics from kernel and per NUMA node"
+}
+
+func (mem *Hugepages) SampleConfig() string {
+	return hugepagesSampleConfig
+}
+
+func (mem *Hugepages) Gather(acc telegraf.Accumulator) error {
+	mem.loadPaths()
+	err := mem.GatherStatsPerNode(acc)
+	if err != nil {
+		return err
+	}
+
+	err = mem.GatherStatsFromMeminfo(acc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GatherHugepagesStatsPerNode collects hugepages stats per NUMA nodes
+func (mem *Hugepages) GatherStatsPerNode(acc telegraf.Accumulator) error {
+	numaNodeMetrics, err := statsPerNUMA(mem.NUMANodePath)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range numaNodeMetrics {
+		metrics := make(map[string]interface{})
+		tags := map[string]string{
+			"node": k,
```

Comment:
what's the possible size of the number of unique occurences of "k"? (ie, is there a cardinality concern here?)

Reply:
Theoretically one might have up to 1023 (max number of NUMA nodes in Linux kernels these days). In the real world it will probably always be <10.